Userland/Libraries/LibWeb/FileAPI/Blob.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2022, Kenneth Myhra <kennethmyhra@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/NonnullRefPtr.h>
+#include <AK/RefCounted.h>
+#include <AK/String.h>
+#include <AK/Vector.h>
+#include <LibWeb/Bindings/BlobWrapper.h>
+#include <LibWeb/Bindings/WindowObject.h>
+#include <LibWeb/Bindings/Wrappable.h>
+#include <LibWeb/Forward.h>
+#include <LibWeb/HTML/Window.h>
+
+namespace Web::FileAPI {
+
+using BlobPart = Variant<JS::Handle<JS::Object>, NonnullRefPtr<Blob>, String>;
+
+struct BlobPropertyBag {
+    String type = "";
+    Bindings::EndingType endings;
+};
+
+class Blob
+    : public RefCounted<Blob>
+    , public Weakable<Blob>
+    , public Bindings::Wrappable {
+
+public:
+    using WrapperType = Bindings::BlobWrapper;
+
+    Blob(ByteBuffer const& byte_buffer, String const& type);
+
+    static NonnullRefPtr<Blob> create(HTML::Window&, Optional<Vector<BlobPart>> const& blob_parts = {}, Optional<BlobPropertyBag> const& options = {})
+    {
+        return adopt_ref(*new Blob(blob_parts, options));
+    }
```

Comment:
No need to pass the window object here, all APIs that want to pass one use `create_with_global_object` - ignore it there.